op-heartbeat/service.go

```diff
@@ -56,32 +54,52 @@ func Main(version string) func(ctx *cli.Context) error {
 			syscall.SIGQUIT,
 		}...)
 		<-doneCh
-		cancel()
-		return nil
+		return srv.Close()
 	}
 }
 
-func Start(ctx context.Context, l log.Logger, cfg Config, version string) error {
-	registry := opmetrics.NewRegistry()
+type HeartbeatService struct {
+	pprof, metrics, http *httputil.HTTPServer
+}
+
+func (hs *HeartbeatService) Close() error {
+	var result error
+	if hs.pprof != nil {
+		result = errors.Join(result, hs.pprof.Close())
+	}
+	if hs.metrics != nil {
+		result = errors.Join(result, hs.metrics.Close())
+	}
+	if hs.http != nil {
+		result = errors.Join(result, hs.http.Close())
+	}
+	return result
+}
```

Comment:
following up on the other comment, if the `HTTPServer` has a `Shutdown(context.Context)`, then we could call `Shutdown`s here and make this `Close` a `Shutdown` instead that could be called from the `Main` with some sensible timeout.